app/main.py

```diff
@@ -1,2 +1,141 @@
+from typing import Any, Optional, Iterator
+from dataclasses import dataclass
+
+from app.point import Point
+
+
+@dataclass
+class Node:
+    key: Any
+    value: Any
+    node_hash: int
+
+
 class Dictionary:
-    pass
+    def __init__(self, initial_capacity: int = 8,
+                 load_factor: float = 2 / 3) -> None:
+        self.capacity: int = initial_capacity
+        self.load_factor: float = load_factor
+        self.size: int = 0
+        self.table: list[Optional[Node]] = [None] * self.capacity
+
+    def __setitem__(self, key: Any, value: Any) -> None:
```

Comment:
The `__setitem__` method should handle collisions using linear probing. You can use a while loop to check if the index is occupied and increment the index until an empty slot is found.